src/reconnect-strategy/mqtts.reconnect-strategy.default.ts

```diff
@@ -0,0 +1,40 @@
+import { MqttsReconnectStrategy } from './mqtts.reconnect-strategy';
+import { ConnectError } from '../errors';
+import {promisify} from 'util';
+const sleep = promisify(setTimeout);
+
+export class MqttsReconnectStrategyDefault implements MqttsReconnectStrategy {
+    /**
+     * Internal attempts counter
+     */
+    #attempts = 1;
+    constructor(
+        /**
+         * Maximum attempts amount
+         */
+        private attempts: number = 60,
+        /**
+         * Interval between attempts, milliseconds
+         */
+        private interval: number = 1000,
+    ) {}
+    should(reason?: any) {
+        if (reason instanceof ConnectError) {
+            if (['NotAuthorized', 'UnacceptableProtocolVersion', 'BadUsernameOrPassword'].includes(reason.status)) {
+                return false;
+            }
+        }
+        if (typeof reason === 'string' && ['Soft disconnect', 'Forced disconnect'].includes(reason)) {
```

Comment:
Comparing strings isn't a good idea in this case. These strings could change at any time and this would be broken.